galsim/wfirst/wfirst_detectors.py

```diff
@@ -83,28 +83,53 @@ def applyIPC(img, edge_treatment='extend', fill_value=None):
     exposures (up to {ncoeff}) supplied as the 'prev_exposures' argument.
-
     For more information about persistence, see the docstring for galsim.Image.applyPersistence.
     Unlike that routine, this one does not need the coefficients to be specified. However, the list
-    of previous eight exposures will have to be supplied. Earlier exposures, if supplied, will be
+    of previous {ncoeff} exposures will have to be supplied. Earlier exposures, if supplied, will be
     ignored.
 
+    'fermi' persistence model:
+    Applies the persistence effect to the Image instance by adding the accumullated persistence dark
+    current of previous exposures (up to galsim.wfirst.max_exps) supplied as the
+    'prev_exposures' argument. Unlike galsim.Image.applyPersistence, this one does not use constant
+    coeffiients but a fermi model (galsim.wfirst.fermi_linear) plus a linear tail below half of
+    saturation.
+
     @param img               The Image to be transformed.
-    @param prev_exposures    List of up to {ncoeff} Image instances in the order of exposures, with the
-                             recent exposure being the first element.
-    """.format(ncoeff=len(galsim.wfirst.persistence_coefficients))
+    @param prev_exposures    List of up to {ncoeff} or {max_exps} Image instances in the order of
+                             exposures, with the recent exposure being the first element
+    @param method            The persistence model ('linear' or 'fermi') to be applied.
+                             [default: 'fermi']
+    """.format(ncoeff=len(galsim.wfirst.persistence_coefficients), max_exps=galsim.wfirst.max_exps)
+
     if not hasattr(prev_exposures,'__iter__'):
-        raise TypeError("In wfirst.applyPersistence, prev_exposures must be a list of Image "
-                        "instances")
+        raise TypeError("In wfirst.applyPersistence, prev_exposures must be a list of Image instances")
+
+    if method == 'linear':
+
+        n_exp = min(len(prev_exposures),len(galsim.wfirst.persistence_coefficients))
+        img.applyPersistence(prev_exposures[:n_exp],galsim.wfirst.persistence_coefficients[:n_exp])
+
+    elif method == 'fermi':
+
+        n_exp = min(len(prev_exposures), galsim.wfirst.max_exps)
+        for i in range(n_exp):
+            ### The length of time where slews and pixel resets happen should be specified.
+            ### Now it is simply assumed to be 0 until we get more information.
+            img += galsim.wfirst.fermi_linear(prev_exposures[i].array)/(1.+ 2*i)
 
-    n_exp = min(len(prev_exposures),len(galsim.wfirst.persistence_coefficients))
-    img.applyPersistence(prev_exposures[:n_exp],galsim.wfirst.persistence_coefficients[:n_exp])
+    else:
+        raise ValueError("applyPersistence only accepts 'linear' or 'fermi' methods.")
 
-def allDetectorEffects(img, prev_exposures=(), rng=None, exptime=default_exptime):
+def allDetectorEffects(img, prev_exposures=[], rng=None, exptime=default_exptime):
```

Comment:
Revert this change.  You should never have a list be a default parameter for a function in python.  cf [this](https://docs.python-guide.org/writing/gotchas/) or [this](https://docs.quantifiedcode.com/python-anti-patterns/correctness/mutable_default_value_as_argument.html) for why this is bad.